tests/test_batch.py

```diff
@@ -521,26 +421,15 @@ def test_rebatch_success(
         batch = self.sdk.rebatch(self.batch_id, **filters)
         assert mock_request.last_request.method == "POST"
 
-        if ids := filters.pop("job_ids", None):
-            filters["id"] = ids
-            # Move id to beginning to construct query param string like the sdk does.
-            if isinstance(filters, OrderedDict):
-                filters.move_to_end("id", last=False)
-
-        # Constructing the query param string
-        query_params = "?"
-        for filter, value in filters.items():
-            if isinstance(value, list):
-                for val in value:
-                    query_params += f"{filter}={val}&"
-            else:
-                if isinstance(value, datetime):
-                    value = value.isoformat()
-                query_params += f"{filter}={value}&"
-        # Replace : by %3A for datetimes
-        query_params = query_params.replace(":", "%3A")[:-1]
-
-        # Check that correct url was requested with query params
+        # Converts 'jobs_ids' into 'id' as it is the field name used by the API
+        rebatch_filters = {
+            ("id" if k == "job_ids" else k): v for k, v in filters.items()
+        }
```

Comment:
```suggestion
        if ids := filters.pop("job_ids", None):
            filters["id"] = ids
```
Maybe you can keep this part to avoid iterating over all attributes

Reply:
I decided to keep it because it is one liner allowing to preserve the index of the key, okay it's an iteration for this use case the scope is alright I think, we do no risk any performance issue.